client/utils/currency/index.js

```diff
@@ -122,7 +168,9 @@ export const formatCurrency = (
 	const isNegative = amount < 0;
 	const positiveAmount = isNegative ? -1 * amount : amount;
 	const prefix = isNegative ? '-' : '';
-	const currency = getCurrency( currencyCode, baseCurrencyCode );
+	const currency = useLocaleFormatting
+		? getCurrencyByLocale( currencyCode )
+		: getCurrency( currencyCode, baseCurrencyCode );
```

Comment:
I didn't want to duplicate logic and, at the same time, wanted it to keep the current behavior in all of its calls. So I just added the `useLocaleFormatting` param and used it in the currency preview only.